pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -559,6 +560,21 @@ func (dr *DashboardServiceImpl) ValidateDashboardBeforeSave(ctx context.Context,
 	return isParentFolderChanged, nil
 }
 
+// WaitForSearchQuery waits for the search query to return the expected number of hits.
+// Since US doesn't offer search-after-write guarantees, we can use this to wait after writes until the indexer is up to date.
+func (dr *DashboardServiceImpl) WaitForSearchQuery(ctx context.Context, query *dashboards.FindPersistedDashboardsQuery, maxRetries int, expectedHits int64) error {
```

Comment:
Nit: as this stuct (`DashboardServiceImpl`) is exposed mostly through the interface, I think this should be private for now.